library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -66,31 +77,33 @@ Http::FilterHeadersStatus PlatformBridgeFilter::onHeaders(Http::HeaderMap& heade
   envoy_headers in_headers = Http::Utility::toBridgeHeaders(headers);
   envoy_filter_headers_status result =
       on_headers(in_headers, end_stream, platform_filter_.instance_context);
-  Http::FilterHeadersStatus status = static_cast<Http::FilterHeadersStatus>(result.status);
-  // TODO(goaway): Current platform implementations expose immutable headers, thus any modification
-  // necessitates a full copy. Add 'modified' bit to determine when we can elide the copy. See also
-  // https://github.com/lyft/envoy-mobile/issues/949 for potential future optimization.
-  headers.clear();
-  for (envoy_header_size_t i = 0; i < result.headers.length; i++) {
-    headers.addCopy(
-        Http::LowerCaseString(Http::Utility::convertToString(result.headers.headers[i].key)),
-        Http::Utility::convertToString(result.headers.headers[i].value));
+
+  switch (result.status) {
+  case kEnvoyFilterHeadersStatusContinue:
+    PlatformBridgeFilter::replaceHeaders(headers, result.headers);
+    return Http::FilterHeadersStatus::Continue;
+
+  case kEnvoyFilterHeadersStatusStopIteration:
+    iteration_state_ = IterationState::Stopped;
+    return Http::FilterHeadersStatus::StopIteration;
+
+  default:
+    PANIC("invalid filter state: unsupported status for platform filters");
```

Comment:
Can we change this to NOT_REACHED_GCOVR_EXCL_LINE? We can add a new macro for NOT_REACHED that takes a string if we want to modify the message. Same with everywhere else.

Reply:
Is that strictly correct though? I mean this _could_ be reached - we receive the value from a public API.